packages/wdio-browserstack-service/src/instrumentation/funnelInstrumentation.ts

```diff
@@ -142,3 +143,106 @@ function getReferrer(framework?: string) {
     const fullName = framework ? 'WebdriverIO-' + framework : 'WebdriverIO'
     return `${fullName}/${BSTACK_SERVICE_VERSION}`
 }
+
+const sendEvent = {
+    tcgDown: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestTcgDownResponse', config),
+    invalidTcgAuth: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestInvalidTcgAuthResponseWithUserImpact', config),
+    tcgAuthFailure: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestTcgAuthFailure', config),
+    tcgtInitSuccessful: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestTcgtInitSuccessful', config),
+    initFailed: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestInitFailedResponse', config),
+    tcgProxyFailure: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestTcgProxyFailure', config),
+}
+
+function isProxyError(authResult: any): boolean {
+    return (authResult as BrowserstackHealing.InitErrorResponse).status === 502
+}
+
+function handleProxyError(config: BrowserStackConfig, isSelfHealEnabled: boolean | undefined) {
+    sendEvent.tcgProxyFailure(config)
```

Comment:
Please wrap all fns with try catch.

Reply:
This has not been done here to maintain consistency of the code with other events like this: https://github.com/xxshubhamxx/webdriverio_ai/pull/5/files/8d28a95c97254877ff05b96d441c5163c9657a9a#diff-e8e0586a18ecd13ae8b55f1344a6a17862c81ab63f012f1e43b9fdaf7f3081a7R29

These are being used as fire and forget so shouldn't be an issue.